File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.js
function solution(a, b, c) {
    const arr = Array.from(new Set([a, b, c]));
    
    if (arr.length === 1) {
        return 27 * Math.pow(a, 6)
    } else if (arr.length === 2) {
        return (a + b + c) * (Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2));
    } else {
        return a + b + c;
    }
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int solution(int a, int b, int c) {
        Set<Integer> set = new HashSet<>();
        set.add(a);
        set.add(b);
        set.add(c);
        
        if (set.size() == 1) return (int) (27 * Math.pow(a, 6));
        if (set.size() == 2) return (a + b + c) * (a * a + b * b + c * c);
        return a + b + c;
    }
}